src/Form/SearchSolrForm.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace Drupal\openy\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a form for setting Search Solr server parameters during install.
+ */
+class SearchSolrForm extends FormBase {
+
+  /**
+   * The configuration factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'openy_solr_search';
+  }
+
+  /**
+   * Constructs a new GroupexFormBase object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The configuration factory.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $solr_server_config = $this->configFactory->get('search_api.server.solr_search');
+    $solr_connector_options = [
+      'standard' => $this->t('Standard'),
+      'basic_auth' => $this->t('Basic Auth'),
+    ];
+    $form['#title'] = $this->t('Configure Solr server');
+    $form['connector'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Solr Connector'),
+      '#description' => $this->t('Choose a connector to use for this Solr server.'),
+      '#options' => $solr_connector_options,
+      '#default_value' => $solr_server_config->get('backend_config.connector'),
+      '#required' => TRUE,
+    ];
+    $form['connector_set'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Configure Solr Connector'),
+      '#collapsible' => FALSE,
+      '#collapsed' => FALSE,
+    ];
+
+    $form['connector_set']['scheme'] = [
+      '#type' => 'select',
+      '#title' => $this->t('HTTP protocol'),
+      '#description' => $this->t('The HTTP protocol to use for sending queries.'),
+      '#default_value' => $solr_server_config->get('backend_config.connector_config.scheme'),
+      '#options' => [
+        'http' => 'http',
+        'https' => 'https',
+      ],
+    ];
+
+    $form['connector_set']['host'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Solr host'),
+      '#description' => $this->t('The host name or IP of your Solr server, e.g. <code>localhost</code> or <code>www.example.com</code>.'),
+      '#default_value' => $solr_server_config->get('backend_config.connector_config.host'),
```

Comment:
If you are using config - instead of having getter for all elements of array from config - do a get once and use a variable without having multiple functions calls across the code.

Reply:
Fixed